wireapp/wire-android-reloaded
# fix: reinitiate email input from saved state

----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?
https://wearezeta.atlassian.net/browse/AR-1598

### Issues

When Android 12 user tries to create personal account and enters email and navigating to webview to see TOU, after returning to the app from webview text field is cleared
### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions
I wasn't able to reproduce the issue and others also tried to reproduce it without a luck

I checked the code responsible for that and I don't see any changes after creating the bug ticket that could solve the problem. 

The only improvement/potential fix I see there is to save email and in case of reinitiating ViewModel with a saved state, view model constructor creates a default TextFieldValue with an empty string so that's the only place I found that can cause the problem, but the ViewModel is created much before so it's not destroyed when a user opens webview, so theoretically this change changes nothing but on the other hand, email was cleared when it shouldn't be. 

also as no one can reproduce the problem maybe it was some weird one time issue, when maybe device memory was full or something and it's an edge case, maybe not worth solving and wait to happen it again 


### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:


### Testing

#### Test Coverage (Optional)


#### How to Test
Happens only on Android 12

Steps:
1. Before Login
2. Tap on Create a personal account
3. Continue
4. input a valid email
5. Continue
6. Tap ToC
7. Terms page opens
8. Device back button

Result:
The email field is cleared

Expected:
The email field should not be cleared

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
